CommunityToolkit.Authentication/ProviderManager.cs

```diff
@@ -54,25 +55,41 @@ public IProvider GlobalProvider
                 var oldState = _provider?.State;
                 if (_provider != null)
                 {
-                    _provider.StateChanged -= ProviderStateChanged;
+                    _provider.StateChanged -= OnProviderStateChanged;
                 }
 
                 _provider = value;
 
                 var newState = _provider?.State;
                 if (_provider != null)
                 {
-                    _provider.StateChanged += ProviderStateChanged;
+                    _provider.StateChanged += OnProviderStateChanged;
                 }
 
                 ProviderUpdated?.Invoke(this, _provider);
                 ProviderStateChanged?.Invoke(this, new ProviderStateChangedEventArgs(oldState, newState));
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(GlobalProvider)));
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(GlobalProviderState)));
             }
         }
 
+        /// <summary>
+        /// Gets the ProviderState of the current GlobalProvider instance.
+        /// Use for binding scenarios instead of ProviderManager.Instance.GlobalProvider.State.
+        /// </summary>
+        public ProviderState? GlobalProviderState => GlobalProvider?.State;
```

Comment:
Wondering too if we made `INotifyPropertyChange` on the BaseProvider too for the State Property if it'd provide the same effect as x:Bind would then just be `{x:Bind auth:ProviderManager.GlobalProvider.State}` right?

Though I like having the state exposed here too for simplicity. Since we're ProviderManager should it just be like `UserState` or just `State` for the property name?

Reply:
Unfortunately, it still fails with INotifyPropertyChanged on auth:ProviderManager.Instance.GlobalProvider.State. The problem is that on app startup the bindings may get setup before the GlobalProvider instance has been set, causing said bindings to fail. This is because theProviderManager.GlobalProvider.Instance can be null, meaning we can't register to listen for changes. However, ProviderManager.Instance is a singleton that is always available, so binding to it works great and won't break if the GlobalProvider instance changes.

But I like the name simplification; I'll change it to State.